framework/aster-frame/src/vm/space.rs

```diff
@@ -80,12 +86,26 @@ impl VmSpace {
         Ok(base_addr)
     }
 
-    /// determine whether a vaddr is already mapped
+    /// Determine whether a vaddr is already mapped.
     pub fn is_mapped(&self, vaddr: Vaddr) -> bool {
         let memory_set = self.memory_set.lock();
         memory_set.is_mapped(vaddr)
     }
 
+    /// Determine whether the target vaddr is writable based on the page table.
+    pub fn is_writable(&self, vaddr: Vaddr) -> bool {
+        let memory_set = self.memory_set.lock();
+        let flags = memory_set.flags(vaddr);
+        flags.is_some_and(|flags| flags.contains(PageTableFlags::WRITABLE))
+    }
+
+    /// Determine whether the target vaddr is executable based on the page table.
```

Comment:
```suggestion
    /// Determine whether the target `vaddr` is executable based on the page table.
```

The comments of `is_mapped` and `is_writable` should be fixed, too.